code/config/lib/Loot/Config/Buildable.hs

```diff
@@ -0,0 +1,92 @@
+{- SPDX-License-Identifier: MPL-2.0 -}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE GADTs #-}
+
+module Loot.Config.Buildable
+  (
+  ) where
+
+import Data.Vinyl (Rec (RNil, (:&)))
+import Fmt (Buildable (..), Builder, blockMapF)
+import GHC.TypeLits (KnownSymbol, symbolVal)
+
+import Loot.Config.Record (ConfigKind, ConfigRec, Item (..), ItemKind,
+                           SumSelection, (::+), (::-), (:::), (::<))
+
+class OptionsToBuilderList (k :: ConfigKind) (is :: [ItemKind]) where
+  configToBuilderList :: ConfigRec k is -> [(Text, Builder)]
+
+instance OptionsToBuilderList (k :: ConfigKind) '[] where
+  configToBuilderList RNil = []
+
+undefinedBuilder :: Builder
+undefinedBuilder = build @Text "<undefined>"
+
+instance
+  forall l v is k
+  . ( KnownSymbol l
```

Comment:
One nitpick: here and in the instances that follow the explicit `forall` seems unnecessary, so I'd suggest to remove it.